### PR TITLE
[#123] enable yarn hot reloading

### PR DESCRIPTION
**Pull request by lavocatt**
When developing it's possible to ask to yarn to enter a mode where it will recompile the bundle on the fly and make the web browser refresh whenever the contributor saves the files on their computer.

### CODE REVIEW

File: package.json
Comment (Msarawan) [inline package.json:1]: Hello @lavocatt,
Thank you for your effort in implementing webpack polling for yarn hot reloading. However, as we're currently transitioning from webpack to Express for building our plugin, I've decided to put this PR on hold for now. The webpack polling isn't compatible with the upcoming changes involving ExpressJS.

Comment (lavocatt) [reply package.json:1]: Thanks for your feedback. There's some documentation online to enable hot reloading also for Express JS https://medium.com/@jogarcia/express-live-reload-everything-5ccf7ceedc4e, can we consider enabling the functionality for express JS also?

Comment (Msarawan) [reply package.json:1]: > Thanks for your feedback. There's some documentation online to enable hot reloading also for Express JS https://medium.com/@jogarcia/express-live-reload-everything-5ccf7ceedc4e, can we consider enabling the functionality for express JS also?

That's nice! sure we can.

Comment (RoddieKieley) [reply package.json:1]: @lavocatt @Msarawan What's the current status here? Are the changes below reflective of a node.js express compatible hot reloading configuration or no?

Comment (gaohoward) [reply package.json:1]: Now we are using express js to serve static and api-server. However the static dir is compiled by webpack. See package.json :
`"start-dev": "ts-node --project tsconfig-server.json ./api-server/app.ts ./dist`
You can see the express server is started directly from the source dir ./api-server/app.ts but the static contents point to the compiled dir ./dist. I don't how to hot-reload the static pages?

Comment (Msarawan) [reply package.json:1]: > @lavocatt @Msarawan What's the current status here? Are the changes below reflective of a node.js express compatible hot reloading configuration or no?

As we're in the process of transitioning from webpack to Express for our plugin, I've put the PR on hold. I'll review it and see how we can integrate hot reloading with the Express server.

Comment (Msarawan) [reply package.json:1]: > Now we are using express js to serve static and api-server. However the static dir is compiled by webpack. See package.json : `"start-dev": "ts-node --project tsconfig-server.json ./api-server/app.ts ./dist` You can see the express server is started directly from the source dir ./api-server/app.ts but the static contents point to the compiled dir ./dist. I don't how to hot-reload the static pages?

Yes @gaohoward , After some digging around with this, I found that there are some ways to enable hot reloading for static pages served by Express using webpack. We need to configure webpack-dev-middleware and webpack-hot-middleware with Express. So, I need to do some investigation with that, and then I will update this pull request.

Comment (lavocatt) [reply package.json:1]: @RoddieKieley the changes were valid prior to @howardgao's PR that introduced expressjs, now this solution is outdated. I was postponing working on this in favor of #130 . @Msarawan has proposed to work on a solution involving `nodemon`.  See https://medium.com/@jogarcia/express-live-reload-everything-5ccf7ceedc4e

**Comment by lavocatt** [comment]
closing this PR in favor of https://github.com/artemiscloud/activemq-artemis-self-provisioning-plugin/pull/139

**Pull request closed without merging**